tests/integration_test/grpc_deadline_test.go

```diff
@@ -190,7 +192,121 @@ plans {
 
 		// Place in closure to allow efficient measuring of elapsed time.
 		// Elapsed time is not checked in the test, it's just for debugging.
-		func() {
+		t.Run(tc.desc, func(t *testing.T) {
+			defer utils.Elapsed(fmt.Sprintf("Test (%s):", tc.desc))()
+
+			// For this client, `err` will always be "exit status 1" on failures.
+			// Check for actual error in `resp` instead.
+			resp, err := client.RunGRPCEchoTest(tc.testPlan, s.Ports().ListenerPort)
+
+			if tc.wantErr == "" && err != nil {
+				t.Errorf("Test (%v): Error during running test: want no err, got err (%v)", tc.desc, resp)
+			}
+
+			if tc.wantErr != "" && err == nil {
+				t.Errorf("Test (%v): Error during running test: got no err, want err (%v)", tc.desc, tc.wantErr)
+			}
+
+			if err != nil && !strings.Contains(resp, tc.wantErr) {
+				t.Errorf("Test (%s): failed, got err (%v), expected err (%v)", tc.desc, resp, tc.wantErr)
+			}
+		})
+	}
+}
+
+func TestIdleTimeoutsForGrpcStreaming(t *testing.T) {
+	t.Parallel()
+
+	testData := []struct {
+		desc        string
+		confArgs    []string
+		addDeadline time.Duration
+		wantErr     string
+		testPlan    string
+	}{
+		// Please be cautious about adding too many time-based tests here.
+		// This can slow down our CI system if we sleep for too long.
+		{
+			desc: "When deadline is NOT specified for method, stream idle timeout specified via flag kicks in and the request fails.",
+			confArgs: append([]string{
+				"--stream_idle_timeout=15s",
+			}, utils.CommonArgs()...),
+			wantErr: `stream timeout`,
+			testPlan: `
+plans {
+  echo_stream {
+    call_config {
+      api_key: "this-is-an-api-key"
+    }
+    request {
+      text: "Hello, world!"
+      response_delay: 20
+    }
+    count: 1
+  }
+}`,
+		},
+		{
+			desc:        "When deadline is specified for method, it overrides the global stream idle timeout and the request succeeds.",
+			addDeadline: 25 * time.Second,
+			confArgs: append([]string{
+				"--stream_idle_timeout=15s",
+			}, utils.CommonArgs()...),
+			testPlan: `
+plans {
+  echo_stream {
+    call_config {
+      api_key: "this-is-an-api-key"
+    }
+    request {
+      text: "Hello, world!"
+      response_delay: 20
+    }
+    count: 1
+  }
+}`,
+		},
+		{
+			desc: "When deadline is NOT specified for method, a low stream idle timeout specified via flag is not honored and the request succeeds.",
```

Comment:
Be specific: a stream idle timeout lower than default response timeout(15s)?

Reply:
Done